gpustack/worker/collector.py

```diff
@@ -75,9 +76,13 @@ def collect(self, initial: bool = False) -> Worker:  # noqa: C901
             try:
                 gpu_devices = self._detector_factory.detect_gpus()
                 status.gpu_devices = gpu_devices
+            except NvidiaSMIInitException:
+                state_message = (
+                    "Error: Failed to initialize NVML: Unknown Error\n"
+                    "Please ensure nvidia-smi is working properly. If you’re running inside Docker, it may be caused by your Docker cgroup driver configuration. More info: <a href=\"https://github.com/NVIDIA/nvidia-container-toolkit/issues/48\">https://github.com/NVIDIA/nvidia-container-toolkit/issues/48</a>"
```

Comment:
Is there any other cases we may get this exception? Do we need a is_docker_env check to expose this info in certain environments